test/extended/util/ldap.go

```diff
@@ -0,0 +1,310 @@
+	if err != nil {
+		return "", nil, err
+	}
+
+	serverDeployment, err := oc.AdminKubeClient().AppsV1().Deployments(oc.Namespace()).Create(deploy)
+	if err != nil {
+		return "", nil, err
+	}
+
+	// Wait for an available replica.
+	err = wait.PollImmediate(1*time.Second, 5*time.Minute, func() (done bool, err error) {
+		dep, getErr := oc.AdminKubeClient().AppsV1().Deployments(oc.Namespace()).Get(serverDeployment.Name,
+			v1.GetOptions{})
+		if getErr != nil {
+			return false, getErr
+		}
+		if dep.Status.AvailableReplicas == 0 {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return "", nil, fmt.Errorf("replica for %s not avaiable: %v", serverDeployment.Name, err)
+	}
+
+	// Confirm ldap server availability.
+	if err := checkLDAPConn(oc, serviceHost); err != nil {
+		return "", nil, err
+	}
+
+	return serviceHost, caPEM, nil
+}
+
+// Confirm that the ldapserver host is responding to ldapsearch.
+func checkLDAPConn(oc *CLI, host string) error {
+	compareString := expectedLDAPClientResponse
+	output, err := runLDAPSearchInPod(oc, host)
+	if err != nil {
+		return err
+	}
+	if !strings.Contains(output, compareString) {
+		return fmt.Errorf("ldapsearch output does not contain %s\n Output: \n%s", compareString, output)
+	}
+	return nil
+}
+
+// Run an ldapsearch in a pod against host.
+func runLDAPSearchInPod(oc *CLI, host string) (string, error) {
+	mounts := []corev1.VolumeMount{
+		{
+			Name:      configMountName,
+			MountPath: configMountPath,
+		},
+		{
+			Name:      certMountName,
+			MountPath: certMountPath,
+		},
+	}
+	volumes := []corev1.Volume{
+		{
+			Name: certMountName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: certMountName,
+				},
+			},
+		},
+		{
+			Name: configMountName,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: configMountName,
+					},
+				},
+			},
+		},
+	}
+	output, err := RunOneShotCommandPod(oc, openldapTestImage, fmt.Sprintf(ldapSearchCommandFormat, host), mounts,
+		volumes, 5*time.Minute)
+	if err != nil {
+		return "", err
+	}
+	return output, nil
+}
+
+func podHasCompleted(pod *corev1.Pod) bool {
+	return len(pod.Status.ContainerStatuses) > 0 &&
+		pod.Status.ContainerStatuses[0].State.Terminated != nil &&
+		pod.Status.ContainerStatuses[0].State.Terminated.Reason == "Completed"
+}
+
+// RunOneShotCommandPod uns the given command in a pod and waits for completion and log output for the given timeout
+// duration, returning the command output or an error.
+func RunOneShotCommandPod(oc *CLI, image, command string, volumeMounts []corev1.VolumeMount, volumes []corev1.Volume,
+	timeout time.Duration) (string, error) {
+	cmd := strings.Split(command, " ")
+	args := cmd[1:]
+	var output string
+
+	pod, err := oc.AdminKubeClient().CoreV1().Pods(oc.Namespace()).Create(newCommandPod(cmd[0], image, cmd[0], args,
```

Comment:
It might be worth extending `test/extended/util/framework.go::NewPodExecutor()` for this functionality?